### PR TITLE
ci: support uploading multiple packages

### PR DESCRIPTION
**Pull request by isabelroses**

### CODE REVIEW

File: .github/workflows/updpkgsums.yml [hunk 1]
```diff
@@ -7,33 +7,50 @@ on:
...
-      - name: Find updated package
+      - name: Find updated packages and construct matrix
+        id: matrix
         run: |
           #!/usr/bin/env bash
           set -euxo pipefail
 
-          echo "pkgbuild=$(git diff --name-only origin/main origin/${GITHUB_HEAD_REF} "*PKGBUILD" | head -1 | xargs dirname)" >> $GITHUB_ENV
+          dirs=$(git diff --name-only HEAD HEAD~1 "*/.SRCINFO" | xargs dirname)
+          # Convert dirs that have been changes to a JSON array
```
Comment (sgoudham) [inline .github/workflows/updpkgsums.yml:34]:
```suggestion
          # Convert dirs that have been changed to a JSON array
```

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -9,27 +9,41 @@ on:
...
-      - name: Publish package
+    steps:
+      - name: Publish packages
         uses: KSXGitHub/github-actions-deploy-aur@v2.7.1 # v2.7.1
-        if: ${{ env.pkgbuild != '' }}
+        if: ${{ matrix.pkgbuild != '' }}
         with:
-          pkgname: packages/${{ env.pkgbuild }}
-          pkgbuild: packages/${{ env.pkgbuild }}/PKGBUILD
+          pkgname: packages/${{ matrix.pkgbuild }}
```
Comment (sgoudham) [inline .github/workflows/publish.yml:45]:
```suggestion
          pkgname: ${{ matrix.pkgbuild }}
```

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -9,27 +9,41 @@ on:
...
+          pkgbuild: packages/${{ matrix.pkgbuild }}/PKGBUILD
```
Comment (sgoudham) [inline .github/workflows/publish.yml:46]: We need to remove `packages` here right?
```suggestion
          pkgbuild: ${{ matrix.pkgbuild }}/PKGBUILD
```

File: .github/workflows/updpkgsums.yml [hunk 4]
```diff
@@ -7,33 +7,50 @@ on:
...
-      - name: Validate package
-        if: ${{ env.pkgbuild != '' }}
+    steps:
+      - name: Validate packages
+        if: ${{ matrix.pkgbuild != '' }}
         uses: ./.github/actions/aur
         with:
           action: 'updpkgsums'
-          pkgname: ${{ env.pkgbuild }}
+          pkgnames: ${{ matrix.pkgbuilds }}
```
Comment (sgoudham) [inline .github/workflows/updpkgsums.yml:50]: I don't see any changes in this PR to accommodate for the rename `pkgname` -> `pkgnames`? I think I might be misunderstanding this.

**Review by sgoudham — CHANGES_REQUESTED** [review]

File: .github/workflows/updpkgsums.yml (hunk 4 above)
Comment (isabelroses) [reply .github/workflows/updpkgsums.yml:50]: Ops your right, i reverted that change.

File: .github/workflows/publish.yml (hunk 3 above)
Comment (isabelroses) [reply .github/workflows/publish.yml:46]: Yeah i think your right

**Review by sgoudham — APPROVED** [review]

**Pull request merged**